src/decorators/index.ts

```diff
@@ -19,19 +19,26 @@
 
 import { Request, Response, RequestHandler, NextFunction } from 'express';
 import jwt from 'jsonwebtoken';
+import fetch from 'node-fetch';
 import { loggerFor } from '../logger';
 import { config } from '../config';
 const L = loggerFor(__filename);
 
-const getToken = (request: Request) => {
+const getToken = async (request: Request) => {
```

Comment:
I suggest changes below to `hasScopes` that require us to provide just the scopes form the token. To handle that, I would change the return type of this method from `Promise<any>` to `Promise<TokenValidationResult>` which uses some new types:

```TypeScript
type Result<T> = {success: false, message: string} | {success: true, data: T};
type TokenData = {
  scopes: string[];
  // ... more token properties if required
}
type TokenValidationResult = Result<TokenData>;
```

Note that `Result<T>` is a useful type that should probably be added in the utils file.

Reply:
You'll also need to modify the two verify methods (`verifyJwt`, `verifyEgoApiKey`) to return just the scopes array.

Reply:
Updated